pytorch_forecasting/metrics/tests/test_all_metrics.py

```diff
@@ -0,0 +1,132 @@
+"""Automated test for all metrics in PyTorch Forecasting."""
+
+from inspect import isclass
+import shutil
+
+import numpy as np
+import pandas as pd
+import pytest
+import torch
+from torch.nn.utils import rnn
+
+from pytorch_forecasting._registry import all_objects
+from pytorch_forecasting.data import TorchNormalizer
+from pytorch_forecasting.tests.test_all_estimators import (
+    BaseFixtureGenerator,
+    PackageConfig,
+)
+
+
+class TestAllPtMetrics(PackageConfig, BaseFixtureGenerator):
+    """Test suite for all metrics in PyTorch Forecasting."""
+
+    object_type_filter = "metric"
+
+    fixture_sequence = ["object_pkg", "object_class"]
+
+    def _test_integration_metrics(self, metric, y_pred, y):
+        from pytorch_forecasting.metrics import PoissonLoss, QuantileLoss
+
+        metric.reset()
+        metric.update(y_pred, y)
+        res = metric.compute()
+        assert isinstance(res, torch.Tensor)
+        assert torch.isfinite(res).all(), "Non-finite values in metric result."
+
+        point_pred = metric.to_prediction(y_pred)
+        assert isinstance(point_pred, torch.Tensor), "Prediction should be a tensor."
+        assert (
+            point_pred.shape[:2] == y_pred.shape[:2]
+        ), "Prediction shape mismatch with y_pred."  # noqa: E501
+        assert point_pred.ndim == 2
+
+        quantiles = [0.1, 0.5, 0.9]
+        if isinstance(metric, QuantileLoss):
```

Comment:
this if/else seems problematic - can you try to replace this by tags related to the expected shape somehow?